pkg/oci/blob_handler.go

```diff
@@ -0,0 +1,160 @@
+package oci
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	extractTimeout = 1 * time.Minute
+)
+
+// Extracts tar.gz files to a specified destination.
+// It takes an io.Reader for the gzip stream and the destination path.
+func (c *Controller) extractTarGz(gzipStream io.Reader, dest string) error {
```

Comment:
So currently our test artifacts contain cluster logs in `*.gz` format. Would it make sense to make this tool capable of extracting compressed files? It would be very useful for the artifact browser.

Reply:
added a new flag!